src/convert/mapEvent.js

```diff
@@ -1,29 +1,44 @@
-const Y = require('yjs');
-const { toSlatePath } = require('../utils/convert');
+const Y = require("yjs");
+const { Block } = require("slate");
+const { toSlatePath } = require("../utils/convert");
 
-// type MapAction = { action: 'add' | 'update' | 'delete'; oldValue: any };
-// type SetNodeOperationProperties = Pick<
-//   SetNodeOperation,
-//   'newProperties' | 'properties'
-// >;
+const mapInsertNodeOperations = (event) => {
+  const changes = Array.from(event.changes.keys.entries());
+  const operations = [];
+  changes.forEach(([key]) => {
+    if (key === "document") {
+      const documentNodes = event.target.get(key).toJSON();
+      const index = 0;
```

Comment:
index is never getting incremented here -- can it be replaced at the point of use by the constant zero, or should it be getting incremented somewhere?

as things stand, I think the elements of operations[] end up in the same order as the corresponding elements of documentNodes[] -- but then if the operations get processed in order, then after the insertions have happened, the inserted nodes will be in reverse order? (entirely possible that I'm misunderstanding something here.)

Reply:
Oh! Good catch! I meant to have it incremented on every pass, got distracted with another issue, then the test passed (because it has only one block) and I fooled myself haha.

Will update the code and test to make sure the index is being correctly incremented.